src/framework/services/scroll/ScrollPosition.js

```diff
@@ -0,0 +1,65 @@
+
+import ThrottledEventDispatcher from '../event/ThrottledEventDispatcher';
+
+export default class ScrollPosition {
+
+  constructor() {
+    this.listeners = [];
+    this.previous = this.current = 0;
```

Comment:
What value should we use? We don't know the distance from the bottom because we don't have a reference to an element yet, and we can't say `0` because that would trigger a lazy-loading call in relevant situations.